### ISSUE TITLE
Add Signer to Api

### ISSUE THREAD

**Issue by ianhe8x** [polkadot-js/api#660]
As a dapp, it doesn't always have access to key pair. Key pairs could be managed in a browser extension or hardware wallet. So I propose to introduce signer to api and let it do the signing. This will allow our browser extension to work with the api by exposing a signer into browser's global.

```
export interface Signer {
  sign(extrinsic: Extrinsic, opt: {from: string} & SignatureOptions): Promise<void>;
}
```

signAndSend is the place to use the signer. I prefer to changing the signature to below which looks more neat
```
  signAndSend (options: SignAndSendOptions)
  signAndSend (options: SignAndSendOptions, statusCb: StatusCb)
  signAndSend (options: SignAndSendOptions, statusCb?: StatusCb)

interface SignAndSendOptions extends Partial<SignatureOptions> {
  from: string | KeyringPair
}
```
though to keep backward compatible, it could also be like:
```
signAndSend (account: KeyringPair | string, _options?: Partial<SignatureOptionsPartial> | StatusCb, statusCb?: StatusCb)
```

I would like to hear from you if it's the right direction before submitting my PR.

**Comment by jacogr** [comment]
Fort going to comment on the signatures: they certainly a bit messy atm, we are not quite happy, but it is moving towards being more useful. We just always try to keep backwards compat, so happy with changes that cleans up the code as long as it does not affect downstream users. 

Honestly, have not considered the signers up to now, which is certainly a gap. (The apps UI manages this by putting stuff on a queue when sent)

So happy to look at any suggestions in this area, it is certainly something that needs to be addressed. 

Would almost suggest leaving the existing (for keypair managed node), adding clean sigs for the new signer and then as we go deprecating the current. (Since the current can be done with a new signer interface). Also so keep the sign/send split as separate steps in mind. (It is useful in some cases)

TL;DR good idea, we certainly need support sooner rather than later

**Comment by jacogr** [comment]
We could keep the signature as you suggested as a last point - would just allow AccountId and Address as well. That would work quite well, is nicely compatible and named as well.

**Comment by jacogr** [comment]
Closed in #662
